massdriver-application-aws-ecs-service/ingress.tf

```diff
@@ -0,0 +1,88 @@
+locals {
+  // Flatten the list of ingresses in the ECS service config
+  container_ingresses = flatten([
+    for container in var.containers : [
+      for port in container.ports : [
+        for ingress in port.ingresses : {
+          container_name = container.name
+          container_port = port.container_port
+          hostname       = ingress.hostname
+          path           = ingress.path
+          # strip the first subdomain block so we are left with just the domain
+          domain = trimprefix(ingress.hostname, "${element(split(".", ingress.hostname), 0)}.")
+        }
+      ]
+    ]
+  ])
+  // Extract out the set of domains desired in the ECS service to lookup in Route53
+  container_ingress_domains = toset(distinct([for ingress in local.container_ingresses : ingress.domain]))
+  // Create a map with the unique pairings of container and port. This is needed for creating and associating target groups
+  container_ingress_port_map = { for ingress in local.container_ingresses :
+    "${ingress.container_name}-${ingress.container_port}" => {
+      container_name = ingress.container_name
+      container_port = ingress.container_port
+      domain         = ingress.domain
+    }
+  }
+
+  // Flatten the ALB listeners into a list of objects for each variation of ingress, listener and domain to make it more easily searchable
+  cluster_listeners = flatten([
+    for ingress in var.ecs_cluster.data.capabilities.ingress : [
+      for listener in ingress.listeners : [
+        for domain in listener.domains : {
+          load_balancer_arn  = ingress.load-balancer-arn
+          security_group_arn = ingress.security-group-arn
+          listener_arn       = listener.arn
+          port               = listener.port
+          protocol           = listener.protocol
+          domain             = domain
+        }
+      ]
+    ]
+  ])
+
+  // The ECS service needs to key off of the ECS ingress (ALB) domains. Iterate through the list of cluster listeners and create
+  // a map of domain -> HTTPS listener that matches the domain, port and protocol
+  domain_to_https_cluster_listener_map = {
+    for listener in local.cluster_listeners : listener.domain => element(flatten([
+      for domain_check in [
+        for port_check in [
+          for protocol_check in local.cluster_listeners : protocol_check if protocol_check.protocol == "https"
+        ] : port_check if port_check.port == 443
+      ] : domain_check if domain_check.domain == listener.domain
+    ]), 0)
+  }
+}
+
+resource "aws_lb_target_group" "service" {
```

Comment:
Super non-blocking but any reason we wouldn't want to use `main` here? I think this resource is unique to the bundle. This name is clearer and you know where I'm coming from. Fine either way. 👍

Reply:
Copy-pasta from testing. I can change to `main`.